src/Controller/KeycloakController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Controller;
+
+use App\Dto\KeycloakUserValidateDto;
+use App\Entity\Domain;
+use App\Entity\User;
+use App\Handler\UserAuthenticationHandler;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapQueryParameter;
+use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
+use Symfony\Component\Routing\Annotation\Route;
+
+class KeycloakController extends AbstractController
+{
+    public function __construct(private readonly EntityManagerInterface $manager, private readonly UserAuthenticationHandler $handler)
+    {}
+
+    #[Route(path: '/api/keycloak', name: 'api_keycloak_index', methods: ['GET'], stateless: true)]
+    public function getUsersSearch(
+        #[MapQueryParameter] string $domain,
+        #[MapQueryParameter] string $search = '',
+        #[MapQueryParameter] int $max = 10,
+        #[MapQueryParameter] int $first = 0,
+    ): Response
+    {
+        if (null === $domainObject = $this->manager->getRepository(Domain::class)->findByName($domain)) {
+            return $this->json([], Response::HTTP_UNPROCESSABLE_ENTITY);
+        }
+
+        $users = $this->manager->getRepository(User::class)->findUsersByString($domainObject, $search, $max, $first)->map(function (User $user) {
+            return [
+                'id' => explode('@', $user->getEmail())[0],
+                'email' => $user->getEmail(),
+            ];
+        });
+        return $this->json($users);
+    }
```

Comment:
Could let symfonys entity resolver handle fetching the domain object or return 404 if it does not exist

```suggestion
    #[Route(path: '/api/keycloak/{domainUrl}', name: 'api_keycloak_index', methods: ['GET'], stateless: true)]
    public function getUsersSearch(
        #[MapEntity(mapping: ['domainUrl' => 'name'])] Domain $domainObject,
        #[MapQueryParameter] string $search = '',
        #[MapQueryParameter] int $max = 10,
        #[MapQueryParameter] int $first = 0,
    ): Response {
        $users = $this->manager->getRepository(User::class)->findUsersByString($domainObject, $search, $max, $first)->map(function (User $user) {
            return [
                'id' => explode('@', $user->getEmail())[0],
                'email' => $user->getEmail(),
            ];
        });
        return $this->json($users);
    }
```